OpenSim/Simulation/Model/Point.h

```diff
@@ -0,0 +1,139 @@
+ *                              OpenSim:  Point.h                             *
+ * -------------------------------------------------------------------------- *
+ * The OpenSim API is a toolkit for musculoskeletal modeling and simulation.  *
+ * See http://opensim.stanford.edu and the NOTICE file for more information.  *
+ * OpenSim is developed at Stanford University and supported by the US        *
+ * National Institutes of Health (U54 GM072970, R24 HD065690) and by DARPA    *
+ * through the Warrior Web program.                                           *
+ *                                                                            *
+ * Copyright (c) 2005-2016 Stanford University and the Authors                *
+ * Author(s): Ajay Seth                                                       *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+// INCLUDE
+#include <OpenSim/Simulation/osimSimulationDLL.h>
+#include <OpenSim/Simulation/Model/ModelComponent.h>
+
+namespace OpenSim {
+
+//=============================================================================
+//=============================================================================
+/**
+ * A Point is an OpenSim abstraction for any point location in space. Points
+ * are intended to locate physical structures (such as points of constraints
+ * and points of muscle attachments) as well as embody the results of spatial
+ * calculations. For example, if your system involves contact, you can define
+ * a Point that describes the location of the center-of-pressure as one
+ * element rolls over another.
+ *
+ * A Point provides its location, velocity and acceleration in the Ground frame
+ * as a function of the Model's (SimTK::MultibodySystem's) state, which is
+ * accessible when the state has been realized to the corresponding
+ * SimTK::Stage (i.e. Position, Velocity and Acceleration). 
+ *
+ * @author Ajay Seth
+ */
+class OSIMSIMULATION_API Point : public ModelComponent {
+OpenSim_DECLARE_ABSTRACT_OBJECT(Point, ModelComponent);
+
+public:
+    OpenSim_DECLARE_OUTPUT(location, SimTK::Vec3, getLocationInGround,
+        SimTK::Stage::Position);
```

Comment:
they all should be `get...InGround` since these are the cached values and avoids unnecessary recalculation. That was the mistake I noticed earlier .. will push the update shortly